features/270-next-skycli/api.md

```diff
@@ -0,0 +1,285 @@
+## Feature Overview
+
+In next, we will have cli tools for user to manage the cluster's applications.
+Since we will only have skycli in the first version, skycli should be able to
+do almost all the operations of v1 portal.
+
+The following spec will cover the EE version first.
+
+
+## User scenarios
+
+- Cluster admin can create cluster user
+- Cluster admin assign user to app
+- Cluster user can create app
+- Cluster user can view, update and delete their apps config
+- Cluster user deploy CF to their apps
+- Cluster user can update apps custom domain setting, add new custom domain to app
+- Cluster admin can do everything in the cluster
+
+## Commands
+
+- skycli login (alisa of `skycli auth login`)
+
+- skycli logout (alisa of `skycli auth logout`)
+
+- skycli auth
+    - [skycli auth login](#skycli-auth-login)
+    - [skycli auth logout](#skycli-auth-logout)
+
+- skycli config
+    - [skycli config view](#skycli-config-view)
+    - [skycli config set-app [APP_NAME]](#skycli-config-set-app)
+    - [skycli config set-cluster-server [https://1.2.3.4]](#skycli-config-set-cluster-server)
+
+- skycli user (Admin only, cluster user management)
+    - [skycli user create](#skycli-user-create)
+    - [skycli user list](#skycli-user-list)
+
+- skycli app
+    - [skycli app list](#skycli-app-list)
+    - [skycli app create](#skycli-app-create)
+    - [skycli app add-user [USER_EMAIL]](#skycli-app-add-user)
+    - [skycli app view-config](#skycli-app-view-config)
+    - [skycli app update-config -f [TENANT_CONFIG_YAML_FILE]](#skycli-app-update-config)
```

Comment:
similar question to the above, would we have per repo config?

Reply:
I don't want to store the config file in repo as the config consists of auth gear sensitive config (e.g. master key, token secret... ). So for the update config flow, user should download the config, edit and re-upload the config. Only the cluster will store the state.

Example
```
# 1. If you don't have a copy of the tenant config file, get it from the view command.
$ skycli app view-config > ./tenant-config.yaml

# 2. Update the tenant-config.yaml file

# 3. Apply the change to the app
$ skycli app update-config -f ./tenant-config.yaml

```